skills_review/skills/taxonomy.py

```diff
@@ -0,0 +1,126 @@
+import pathlib
+
+import numpy as np
+import pandas as pd
+
+from sentence_transformers import SentenceTransformer
+
+
+__here__ = pathlib.Path(__file__).parent
+data_dir = __here__ / ".." / ".."/"data"
+
+nlp_json_file = data_dir / "nlp_generated_skills.json"
+pickle_file = data_dir/"job_title_embeddings.pkl"
+
+assert nlp_json_file.exists()
+
+
+def get_job_embeddings(job_titles):
+    model_name = "all-MiniLM-L6-v2"
+
+    model = SentenceTransformer(model_name)
+    # Sentences are encoded by calling model.encode()
+    embeddings = model.encode(job_titles)
+    embedding_df = pd.DataFrame(index=job_titles, data=embeddings)
+
+    return embedding_df
+
+
+def create_job_title_embeddings():
+    if not pickle_file.exists():
+        skill_sample_size = 10000
+
+        nlp_jobs_df = pd.read_json(nlp_json_file)[["user_id", "job_title"]].iloc[0:skill_sample_size]
+        df = nlp_jobs_df.copy()
+
+        # converts to numpy array for speed
+        unique_job_titles = df.drop_duplicates(subset=["job_title"]).dropna(axis=0)["job_title"].to_numpy()
+        embeddings = get_job_embeddings(unique_job_titles)
+
+        # todo - add to database in some way
+        embeddings.to_pickle(pickle_file)
+
+
+def return_similar_title_skills(job_title, user_skills, job_embeddings):
+    skill_count = 10
+    model_name = "all-MiniLM-L6-v2"
+    missing_ratings = False
+
+    if job_title not in job_embeddings.index.to_list():
+        user_skills = user_skills[user_skills["job_title"] != job_title].copy()
+
+    unique_job_titles = user_skills.drop_duplicates(subset=["job_title"]).dropna(axis=0)["job_title"].to_numpy()
+
+    if missing_ratings:
```

Comment:
Can this ever be true?

Reply:
Just to be clear.  That was Andreas' code not mine